engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -197,101 +266,97 @@ public void render(DefaultRenderingProcess.StereoRenderState stereoRenderState)
         switch (stereoRenderState) {
             case MONO:
                 currentRenderingStage = WorldRenderingStage.DEFAULT;
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_LEFT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_LEFT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_RIGHT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_RIGHT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = false;
                 break;
         }
 
-        if (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE) {
```

Comment:
This check is really all about making sure some things happen only once per frame, hence the new isFirstRenderingStageForCurrentFrame variable.